subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/processing/AggregatingProcessingStrategy.java

```diff
@@ -67,4 +69,13 @@ private void recordAggregatedTypes(Set<String> supportedAnnotationTypes, Set<? e
     public void recordGeneratedType(CharSequence name, Element[] originatingElements) {
         result.getGeneratedAggregatingTypes().add(name.toString());
     }
+
+    @Override
+    public void recordGeneratedResource(JavaFileManager.Location location, CharSequence pkg, CharSequence relativeName, Element[] originatingElements) {
+        try {
+            result.getGeneratedAggregatingResources().add(new GeneratedResource(GeneratedResource.Location.from(location), pkg, relativeName));
+        } catch (IllegalArgumentException e) {
```

Comment:
I'd prefer not using exceptions here, but instead having the `Location.from` method return null if that location is not supported and then do a null-check here. Otherwise we would significantly hurt performance in case some processor actually uses an unsupported location.

Reply:
Will fix.